core/citrus-base/src/main/java/org/citrusframework/DefaultTestCaseRunner.java

```diff
@@ -151,4 +151,18 @@ public void setTestCase(TestCase testCase) {
         this.testCase = testCase;
         this.testCase.setIncremental(true);
     }
+
+    public static class DefaultTestCaseRunnerProvider implements TestCaseRunnerProvider {
+
+        @Override
+        public TestCaseRunner createTestCaseRunner(TestContext context) {
+            return new DefaultTestCaseRunner(context);
```

Comment:
I think the constructor of `DefaultTestCaseRunner` should be `private`? otherwise we still encourage users to use it directly.

Reply:
The DefaultTestCaseRunner is directly instantiated in hundrets of testcases. Therefore I have not done this.

Reply:
ok. I mean, people _can_ still use it.